### ISSUE TITLE
Support JSON directory listing

### ISSUE THREAD

**Issue by jonasgeiler** [joseluisq/static-web-server#128]
**Is your feature request related to a problem? Please describe.**
I am trying to write a program that needs to interact with a remote directory listing, delivered by static-web-server. Currently I am parsing the HTML site and extract the required info that way, but it would be very useful if I could receive that directory listing in JSON format instead.  
The nginx directory listing supports something like this, by setting `autoindex_format` to `json`, but I'd rather use static-web-server instead of nginx...

**Describe the solution you'd like**
I would like to be able to set a `format` parameter when listing the contents of a directory, that allows setting the returned format to either `html` (default) or `json`, e.g. `https://localhost/media/?format=json`.  
The JSON could look like this (taken from nginx):
```json
[
  {
    "name": "videos",
    "type": "directory",
    "mtime": "2022-04-18T22:44:01Z"
  },
  {
    "name": "image.jpg",
    "type": "file",
    "mtime": "2022-04-18T23:44:01Z",
    "size": 14052802
  }
]
```

**Describe alternatives you've considered**
As said above, I tried using nginx, which was wayy to complex for me, and now I am parsing the HTML directory listing. Both of these solutions aren't very "elegant".

**Comment by joseluisq** [comment]
Released on [v2.13.0](https://github.com/joseluisq/static-web-server/releases/tag/v2.13.0)

**Comment by jonasgeiler** [comment]
Thank you so much!
